samza-core/src/main/java/org/apache/samza/runtime/AbstractApplicationRunner.java

```diff
@@ -26,7 +26,7 @@
 
 public abstract class AbstractApplicationRunner implements ApplicationRunner {
```

Comment:
> Abstract classes and interfaces are not mutually exclusive. For example see:

I didn't imply they are mutually exclusive. I am just saying it is a pattern that should be used with care. Whenever possible, we should simplify it. 

>However, I will take a second look at this case tomorrow, to ensure that the interface is still useful

Thank you!